### PR TITLE
CSPL-2036, CSPL-2037: [Azure AFW] [QA] Edit testenv folder, app framework,LM,secret tests to have them run on Azure

### PR DESCRIPTION
**Pull request by jambrosiano**
This PR adds the support to run existing app framework, secret, smartstore, License Master and License Manager tests on Azure platform.
The appframework tests are now split between 2 folders, 1 for aws, 1 for azure, and the files inside will use specific utils necessary to perform apps operation with S3 or Azure blob.
It also contains additional utils necessary for tests to run on Azure.
For secret, License Master and License Manager, tests are still in the same folders, each tests needing S3 before are updated to be able to use S3 or Azure depending or where they are run.
There's also an update on the README.md mentioning github actions and Azure, previous version was still referring to CircleCI

### CODE REVIEW

File: test/appframework/aws/m4/appframework_aws_suite_test.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ var (
 	currDir, _            = os.Getwd()
 	downloadDirV1         = filepath.Join(currDir, "m4appfwV1-"+testenv.RandomDNSName(4))
 	downloadDirV2         = filepath.Join(currDir, "m4appfwV2-"+testenv.RandomDNSName(4))
-	downloadDirPVTestApps = filepath.Join(currDir, "m4appfwPVTestApps-"+testenv.RandomDNSName(4))
+	downloadDirPVTestApps = filepath.Join(currDir, "c3appfwPVTestApps-"+testenv.RandomDNSName(4))
```
Comment (tpereirasplunk) [inline test/appframework/aws/m4/appframework_aws_suite_test.go:53]: NIT: should the names not be changed. This is the m4 directory

File: test/appframework/aws/s1/appframework_aws_suite_test.go [hunk 2]
```diff
@@ -50,7 +50,7 @@ var (
 	currDir, _            = os.Getwd()
 	downloadDirV1         = filepath.Join(currDir, "s1appfwV1-"+testenv.RandomDNSName(4))
 	downloadDirV2         = filepath.Join(currDir, "s1appfwV2-"+testenv.RandomDNSName(4))
-	downloadDirPVTestApps = filepath.Join(currDir, "s1appfwPVTestApps-"+testenv.RandomDNSName(4))
+	downloadDirPVTestApps = filepath.Join(currDir, "c3appfwPVTestApps-"+testenv.RandomDNSName(4))
```
Comment (tpereirasplunk) [inline test/appframework/aws/s1/appframework_aws_suite_test.go:53]: NIT: should the names be `m4appfwPVTestApps `. This is the m4 directory

File: test/appframework/aws/m4/appframework_aws_suite_test.go (hunk 1 above)
Comment (jambrosiano) [reply test/appframework/aws/m4/appframework_aws_suite_test.go:53]: Indeed, bad copy-paste

File: test/appframework/aws/s1/appframework_aws_suite_test.go (hunk 2 above)
Comment (jambrosiano) [reply test/appframework/aws/s1/appframework_aws_suite_test.go:53]: Indeed, bad copy-paste

**Review by kumarajeet — COMMENTED** [review]
My comments are more likely towards code cleanup (some logging change and removal of code not used).

File: test/testenv/azureutils.go [hunk 3]
```diff
@@ -404,32 +299,358 @@ func setUploadHeaders(ctx context.Context, fileName string, request *http.Reques
...
+			logf.Log.Error(err, "Unable to upload file", "File name", key)
+			return nil, err
+		}
+		uploadedFiles = append(uploadedFiles, fileName)
+	}
+	return uploadedFiles, nil
+}
+
+// ListFilesonAzure list the files present in an Azure Storage account container
+func ListFilesonAzure(ctx context.Context, accountName, accountKey, containerName, containerPath string) {
```
[the remark this answers is not in the document]
Comment (jambrosiano) [reply test/testenv/azureutils.go:445]: Will keep it for now

**Review by kumarajeet — APPROVED** [review]

**Pull request merged**